spec/controllers/social_networking/profile_pages_controller_spec.rb

```diff
@@ -3,34 +3,76 @@
 module SocialNetworking
   describe ProfilePagesController, type: :controller do
     let(:group) { double("group", social_networking_profile_questions: []) }
-    let(:participant_status) { double("participant_status", :context= => "") }
     let(:participant) do
-      double(
-        "participant",
+      instance_double(
+        Participant,
         active_group: group,
-        navigation_status: participant_status)
+        id: 1)
     end
-    let(:initiator) { double("participant") }
+    let(:active_profile) do
+      instance_double(
+        Profile,
+        participant_id: 1,
+        active: true)
+    end
+    let(:profile) { instance_double(Profile, participant_id: 1, active: nil) }
+    let(:initiator) { instance_double(Participant) }
+    let(:nudge) { instance_double(Nudge, initiator: initiator) }
 
     describe "GET show" do
       context "when the current participant is authenticated" do
         before do
           @routes = Engine.routes
-          allow(controller).to receive(:authenticate_participant!)
           allow(controller).to receive(:current_participant) { participant }
+
+          expect(controller).to receive(:authenticate_participant!)
+          expect(controller).to receive(:set_member_profiles)
+          expect(controller).to receive(:load_feed_items)
         end
 
         context "when params contains profile id" do
+          context "current participant gets nudged" do
+            it "displays the display name of the participant who nudged" do
+              expect(Profile).to receive(:find) { profile }
+              expect(Nudge).to receive(:search) { [nudge] }
+              expect(initiator).to receive(:display_name)
+
+              get :show, id: 1
+            end
+          end
+        end
+
+        context "when viewing personal profile" do
           after do
-            get :show, id: 1
+            get :show
```

Comment:
I'd love to see `get :show` repeated in each spec instead of in this `after` block. I think it's more readable that way.